cpp11/Ice/interceptor/Server.cpp

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+#include <unordered_set>
+#include <Ice/Ice.h>
+#include <AuthenticatorI.h>
+#include <InterceptorI.h>
+#include <ThermostatI.h>
+
+using namespace std;
+
+int main(int argc, char* argv[])
+{
+    int status = 0;
+
+    try
+    {
+        //
+        // CtrlCHandler must be created before the communicator
+        // or any other threads are started.
+        //
+        Ice::CtrlCHandler ctrlCHandler;
+
+        //
+        // CommunicatorHolder's ctor initializes an Ice communicator,
+        // and its dtor destroys this communicator.
+        //
+        Ice::CommunicatorHolder ich(argc, argv, "config.server");
+        auto communicator = ich.communicator();
+
+        ctrlCHandler.setCallback(
+            [communicator](int)
+            {
+                communicator->shutdown();
+            });
+
+        //
+        // The communicator initialization removes all Ice-related arguments from argc/argv
+        //
+        if(argc > 1)
+        {
+            cerr << argv[0] << ": too many arguments" << endl;
+            status = 1;
+        }
+        else
+        {
+            //
```

Comment:
It might not be immediately clear to the user why we create two object adapters here. I assume it's to allow the configuration of restricted endpoint to access the authenticator object? A comment to explain this would be good, here and also in the server configuration file.